File: lib/components/music_player/generated_player_card_for_session/types/circle_type.dart
enum CircleType {
  primary("primary"),
  secondary("secondary"),
  tertiary("tertiary"),
  none("none");

  final String type;
  const CircleType(this.type);

  /// [canvasRadius] = min(canvas.width, canvas.height)
  ({double minRadius, double maxRadius}) getRadiusRange({required double canvasRadius}) {
    final double baseRadius = canvasRadius / 2;

    return switch (this) {
      primary => (minRadius: baseRadius * 0.7, maxRadius: baseRadius * 0.9),
      secondary => (minRadius: baseRadius * 0.6, maxRadius: baseRadius * 0.8),
      tertiary => (minRadius: baseRadius * 0.5, maxRadius: baseRadius * 0.7),
      none => (minRadius: baseRadius * 0.4, maxRadius: baseRadius * 0.6),
    };
  }
}

File: android/app/src/main/kotlin/com/wavepaths/player/foreground/utils/generateAlbumCover/types/CircleType.kt
package com.wavepaths.player.foreground.utils.generateAlbumCover.types

enum class CircleType(val type: String) {
    PRIMARY("primary"),
    SECONDARY("secondary"),
    TERTIARY("tertiary"),
    NONE("none");

    data class RadiusRange(val min: Double, val max: Double)

    /** [canvasRadius] = min(canvas.width, canvas.height) */
    fun getRadiusRange(canvasRadius: Double): RadiusRange {
        val baseRadius = canvasRadius / 2
        return when (this) {
            PRIMARY -> RadiusRange(baseRadius * 0.7, baseRadius * 0.9)
            SECONDARY -> RadiusRange(baseRadius * 0.6, baseRadius * 0.8)
            TERTIARY -> RadiusRange(baseRadius * 0.5, baseRadius * 0.7)
            NONE -> RadiusRange(baseRadius * 0.4, baseRadius * 0.6)
        }
    }
}
